crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -175,14 +175,23 @@ def openstack_check
         ret = {}
         # swift replicas check vs. number of disks
         prop = Proposal.where(barclamp: "swift").first
-        return ret if prop.nil?
-        replicas = prop["attributes"]["swift"]["replicas"] || 0
+        unless prop.nil?
+          replicas = prop["attributes"]["swift"]["replicas"] || 0
```

Comment:
this change, although needed, is unrelated to this PR. Overall looks good to me.